include/kafka/Error.h

```diff
@@ -62,5 +64,59 @@ struct ErrorWithDetail
     std::string     detail;
 };
 
+/**
+ * Error type with rich info.
+ */
+class Error
+{
+public:
+    explicit Error(rd_kafka_error_t* error = nullptr): _rkError(error, RkErrorDeleter) {}
+    Error(const Error&)            = default;
+    virtual ~Error()               = default;
+
+    /**
+     * Obtains the underlying error code.
+     */
+    virtual std::error_code     errorCode()   const { return ErrorCode(rd_kafka_error_code(_rkError.get())); }
+
+    /**
+     * Readable error string.
+     */
+    virtual std::string         message()     const { return rd_kafka_error_string(_rkError.get()); }
+
+    /**
+     * Fatal error indicates that the client instance is no longer usable.
+     */
+    virtual std::optional<bool> isFatal()     const { return rd_kafka_error_is_fatal(_rkError.get()); }
+
+    /**
+     * Show whether the operation may be retried.
+     */
+    virtual std::optional<bool> isRetriable() const { return rd_kafka_error_is_retriable(_rkError.get()); }
+
+private:
+    rd_kafka_error_shared_ptr _rkError;
+};
+
+/**
+ * Error type only with brief info.
+ */
+class SimpleError: public Error
```

Comment:
Should this SimpleError inherit from Error? I did not see how they are related.

Reply:
Underlying they're the same stuff, one is from the legacy `rd_kafka_resp_err_t`, while the other is from the new `rd_kafka_error_t` which contains the rich-info (including the `rd_kafka_resp_err_t`)

Reply:
And, hopefully, one day the `SimpleError` could just be removed, -- after `librdkafka` migrating all interfaces with `rd_kafka_error_t` (rich-info error type)